yarn-project/simulator/src/avm/journal/public_storage.ts

```diff
@@ -0,0 +1,149 @@
+import { Fr } from '@aztec/foundation/fields';
+
+import type { PublicStateDB } from '../../index.js';
+
+/**
+ * A class to manage public storage reads and writes during a contract call's AVM simulation.
+ * Maintains a storage write cache, and ensures that reads fall back to the correct source.
+ * When a contract call completes, its storage cache can be merged into its parent's.
+ */
+export class PublicStorage {
+  /** Cached storage writes. */
+  private cache: PublicStorageCache;
```

Comment:
I'm open to merging these two classes and just including the cache functionality in PublicStorage. I do kind of like that the cache class is very very simple.

Reply:
I dont think its required, theyre in the same file so the overhead is nil